tests/unit/linalg/test_cg_indefinite.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+#include "catch.hpp"
+
+using namespace mfem;
+
+class FakeSolver : public Solver
+{
+private:
+   Operator& op;
+
+public:
+   FakeSolver(Operator& op_) : Solver(op_.Height()), op(op_) { }
+   void SetOperator(const Operator &op) { }
+   void Mult(const Vector& x, Vector& y) const { op.Mult(x, y); }
+};
+
+TEST_CASE("CGSolver", "[Indefinite]")
+{
+   // Define indefinite SparseMatrix
+   SparseMatrix indefinite(2, 2);
+   indefinite.Add(0, 1, 1.0);
+   indefinite.Add(1, 0, 1.0);
+   indefinite.Finalize();
+
+   Vector v(2);
+   v(0) = 1.0;
+   v(1) = -1.0;
+   Vector x(2);
+   x = 0.0;
+
+   // check indefinite operator
```

Comment:
Indefinite operators are still allowed (in which case only a warning is printed), so this comment is a bit confusing. For example, scaling the operator in example 1 by -1 in the PA case (using a Jacobi preconditioned) results in convergence with warnings about an indefinite operator. It might be better to clarify this test with a comment that more precisely describes how it fails.

Reply:
In my view, CG just doesn't make sense with an indefinite operator. The example you give has a negative definite operator, which is sensible (but a bit unorthodox) for CG. I've tried to clarify the comments to refer to just the example we're checking in the test (which is clearly inappropriate for CG). Let me know if I need to clarify more.